modules/unsupported/jdbc-singlestore/src/main/java/org/geotools/data/singlestore/SingleStoreDialect.java

```diff
@@ -0,0 +1,398 @@
+                }
+            }
+        } finally {
+            dataStore.closeSafe(rs);
+        }
+
+        // create spatial index for all geometry columns
+        for (AttributeDescriptor ad : featureType.getAttributeDescriptors()) {
+            if (!(ad instanceof GeometryDescriptor)) {
+                continue;
+            }
+            GeometryDescriptor gd = (GeometryDescriptor) ad;
+
+            if (!ad.isNillable()) {
+                // can only index non null columns
+                StringBuffer sql = new StringBuffer("ALTER TABLE ");
+                encodeTableName(featureType.getTypeName(), sql);
+                sql.append(" ADD SPATIAL INDEX (");
+                encodeColumnName(null, gd.getLocalName(), sql);
+                sql.append(")");
+
+                LOGGER.fine(sql.toString());
+                Statement st = cx.createStatement();
+                try {
+                    st.execute(sql.toString());
+                } finally {
+                    dataStore.closeSafe(st);
+                }
+            }
+
+            CoordinateReferenceSystem crs = gd.getCoordinateReferenceSystem();
+            int srid = 0;
+            if (crs != null) {
+                Integer i = null;
+                try {
+                    i = CRS.lookupEpsgCode(crs, true);
+                } catch (FactoryException e) {
+                    LOGGER.log(Level.FINER, "Could not determine epsg code", e);
+                }
+                srid = i != null ? i : srid;
+            }
+
+            StringBuffer sql = new StringBuffer("INSERT INTO ");
+            encodeTableName("geometry_columns", sql);
+            sql.append(" (");
+            encodeColumnName(null, "f_table_schema", sql);
+            sql.append(", ");
+            encodeColumnName(null, "f_table_name", sql);
+            sql.append(", ");
+            encodeColumnName(null, "f_geometry_column", sql);
+            sql.append(", ");
+            encodeColumnName(null, "coord_dimension", sql);
+            sql.append(", ");
+            encodeColumnName(null, "srid", sql);
+            sql.append(", ");
+            encodeColumnName(null, "type", sql);
+            sql.append(") ");
+            sql.append(" VALUES (");
+            sql.append(schemaName != null ? "'" + schemaName + "'" : "NULL").append(", ");
+            sql.append("'").append(featureType.getTypeName()).append("', ");
+            sql.append("'").append(ad.getLocalName()).append("', ");
+            sql.append("2, ");
+            sql.append(srid).append(", ");
+
+            @SuppressWarnings("unchecked")
+            Class<? extends Geometry> gc =
+                    (Class<? extends Geometry>) gd.getType().getBinding();
+            Geometries g = Geometries.getForBinding(gc);
+            sql.append("'")
+                    .append(g != null ? g.getName().toUpperCase() : "GEOMETRY")
+                    .append("')");
+
+            LOGGER.fine(sql.toString());
```

Comment:
Even if not critical I would probably put the sql.toString in a variable and use it in both the execute and the logging.